app/javascript/packages/document-capture/components/document-capture.jsx

```diff
@@ -70,10 +82,27 @@ function DocumentCapture({ isLivenessEnabled = true }) {
     setFormValues(nextFormValues);
   }
 
-  const isFormEntriesError = submissionError && submissionError instanceof UploadFormEntriesError;
+  /** @type {string=} */
   let initialStep;
+
+  /** @type {FormStepError[]=} */
+  let initialActiveErrors;
+
+  /** @type {boolean} */
+  let isUnknownError = false;
+
   if (submissionError) {
-    initialStep = isFormEntriesError || !isLivenessEnabled ? 'documents' : 'selfie';
+    if (submissionError instanceof UploadFormEntriesError) {
+      initialActiveErrors = submissionError.rawErrors.map((error) => ({
+        field: error.field,
+        error,
+      }));
+
+      initialStep = getInitialStep(steps, initialActiveErrors);
+    } else {
+      isUnknownError = true;
+      initialStep = steps[steps.length - 1].name;
+    }
```

Comment:
I might want to try to clean this up a bit.

Maybe:

- Extracting a better arrangement of utility functions (e.g. `getInitialStep`, `getInitialActiveErrors`, `isUnknownError`, etc)
- Possibly extracting a separate _component_ for rendering the error path